python/lsst/ts/ataos/ataos_csc.py

```diff
@@ -317,7 +326,11 @@ def mark_corrections(self, data, flag):
         if data.moveWhileExposing:
             self.move_while_exposing = flag
 
-        if data.all:
+        if hasattr(data, "enableAll") and data.enableAll:
+            for key in self.corrections:
+                self.corrections[key] = flag
+            return
+        elif hasattr(data, "disableAll") and data.disableAll:
```

Comment:
Regarding the `publish_enable_corrections` method below: if you are willing to use ts_salobj 3.7 then that code can be simplified to:
```
    def publish_enable_corrections(self):
        """Utility function to publish enable corrections."""
        kwargs = dict((key, value) for key, value in self.corrections.items())
        kwargs["moveWhileExposing"] = self.move_while_exposing
        self.evt_correctionEnabled.set_put(**kwargs)
```

Better yet, if the only event that holds this information is `correctionEnabled` then I personally recommend keeping the data in the event itself (again this requires ts_salobj 3.7) -- replace `self.corrections` with  `self.evt_correctionEnabled.data` basically everywhere.

Reply:
I like the idea of using `self.evt_correctionEnabled.data` directly but, while trying to implement it, it make some of the things I do elsewhere quite bulky. Especially things like this:

https://github.com/lsst-ts/ts_ataos/blob/008131bd2ee133a8eb98b6cf52d8ce900711aa4c/python/lsst/ts/ataos/ataos_csc.py#L289-L294

and 

https://github.com/lsst-ts/ts_ataos/blob/008131bd2ee133a8eb98b6cf52d8ce900711aa4c/python/lsst/ts/ataos/ataos_csc.py#L330-L331